### DISCUSSION TITLE
smartmontools: update-smart-drivedb uses outdated curl version from /usr/bin/curl

### DISCUSSION THREAD

**Discussion by term-i-nator** [Homebrew/discussions#3413] [Writing Formulae/Casks]
### smartmontools: update-smart-drivedb – pull request...?

By default update-smart-drivedb uses the (outdated) mac version of curl under /usr/bin/curl even if the latest version is installed via homebrew (/usr/local/opt/curl/bin/curl). Christian Franke from smartmontools suggested to open a pull request at homebrew to add the line

--with-scriptpath='/usr/local/opt/curl/bin:/usr/local/bin:/usr/bin:/bin'

to 

https://github.com/Homebrew/homebrew-core/blob/master/Formula/smartmontools.rb

I don't know how to open a correct pull request and I don't have the time to learn how to do it now. Could anyone please open a pull request?

Thanks in advance!

**Comment by carlocab** [comment]
Why do you need the latest version of `curl` for `smartmontools`? macOS ships a relatively new version of `curl`, so unless `smartmontools` has very particular requirements, the Apple-provided `curl` should suit.

Does `smartmontools` not check your `PATH` to find `curl`?

**Reply by term-i-nator** [reply]
The macOS (Monterey 12.4) version of `curl` is 7.79.1 while the homebrew version is 7.84.0. It's not that far behind, but as always, it's recommended to use the latest version of any application due to security fixes etc. If there's a newer version installed, why not use it?

In my `PATH` and even in root's `PATH` the homebrew version is prefered (`which curl `=>` /usr/local/opt/curl/bin/curl`), but `update-smart-drivedb` still uses `/usr/bin/curl`. As a temporary fix I added the line `export PATH="/usr/local/opt/curl/bin:/usr/local/bin:/usr/bin:/bin"` at the beginning of the script `/usr/local/Cellar/smartmontools/7.3/bin/update-smart-drivedb`. Christian Franke from `smartmontools` suggested to add the line in `smarmontools.rb` as stated in my first post, so my temporary fix would be permanent.

**Reply by SMillerDev** [reply]
I think a better fix is if the tool respected `$PATH` instead of Homebrew hardcoding a `$PATH` into the application.

**Reply by carlocab** [reply]
Why doesn't `update-smart-drivedb` respect the caller's `PATH` when it runs?

It seems more appropriate to me to fix `update-smart-drivedb` so that it does that, instead of hard-coding a `PATH` that could be wrong or inappropriate for some uses.

Of course, that solution seems preferable to me because I don't understand the reasons why `update-smart-drivedb` determines its `PATH` at build-time instead of at run-time, so finding this out would probably be good.

**Edit:** Oh, this is exactly @SMillerDev's point above. Didn't see it while I was typing my comment.

**Reply by term-i-nator** [reply]
You're right (all of you). Hardcoding a `PATH` is never a good idea. That's why I first contacted the development team of smartmontools. They suggested to insert the line above into `smartmontools.rb`. I have no idea, why `update-smart-drivedb` doesn't respect my `PATH`. I call this script from a weekly script (not cron) under my own account, so my `PATH` should be used.

**Reply by Bo98** [reply]
Looks like we can pass `--with-scriptpath=no` and it should take the `PATH`.

**Comment by samm-git** [comment]
`--with-scriptpath=no` is fore sure will solve topicstarter problem, from security standpoint not sure what is better, may be as an old-school person i prefer to have fixed path. But i am not opiniated on that, whatever option would work.
